modules/mcc/include/opencv2/mcc/ccm.hpp

```diff
@@ -0,0 +1,537 @@
+Take the R-channel as an example:
+
+\f[
+R=\arg min_{f}(\Sigma(R_{dl}-f(R_S)^2)
+\f]
+
+It's equivalent to finding the least square regression for below equations:
+\f[
+f(R_{s1})=R_{dl1}\\
+f(R_{s2})=R_{dl2}\\
+...
+\f]
+
+With a polynomial, the above equations becomes:
+\f[
+\begin{bmatrix}
+R_{s1}^{n} & R_{s1}^{n-1} & ... & 1\\
+R_{s2}^{n} & R_{s2}^{n-1} & ... & 1\\
+... & ... & ... & ...
+\end{bmatrix}
+\begin{bmatrix}
+a_{n}\\
+a_{n-1}\\
+... \\
+a_0
+\end{bmatrix}
+=
+\begin{bmatrix}
+R_{dl1}\\
+R_{dl2}\\
+...
+\end{bmatrix}
+\f]
+It can be expressed as a system of linear equations:
+
+\f[
+AX=B
+\f]
+
+When the number of reference colors is not less than the degree of the polynomial, the linear system has a least-squares solution:
+
+\f[
+X=(A^TA)^{-1}A^TB
+\f]
+
+Once we get the polynomial coefficients, we can get the polynomial r.
+
+This method of finding polynomial coefficients can be implemented by numpy.polyfit in numpy, expressed here as:
+
+\f[
+R=polyfit(R_S, R_{dl})
+\f]
+
+Note that, in general, the polynomial that we want to obtain is guaranteed to monotonically increase in the interval [0,1] ,
+but this means that nonlinear method is needed to generate the polynomials(see [4] for detail).
+This would greatly increases the complexity of the program.
+Considering that the monotonicity does not affect the correct operation of the color correction program, polyfit is still used to implement the program.
+
+Parameters for other channels can also be derived in a similar way.
+
+#### Grayscale Polynomial Fitting
+
+In this method[2], single polynomial is used for all channels.
+The polynomial is still a polyfit result from the detected colors to the linear reference colors.
+However, only the gray of the reference colors can participate in the calculation.
+
+Since the detected colors corresponding to the gray of reference colors is not necessarily gray, it needs to be grayed.
+Grayscale refers to the Y channel of the XYZ color space.
+The color space of the detected data is not determined and cannot be converted into the XYZ space.
+Therefore, the sRGB formula is used to approximate[5].
+\f[
+G_{s}=0.2126R_{s}+0.7152G_{s}+0.0722B_{s}
+\f]
+Then the polynomial parameters can be obtained by using the polyfit.
+\f[
+f=polyfit(G_{s}, G_{dl})
+\f]
+After \f$f\f$ is obtained, linearization can be performed.
+
+#### Logarithmic Polynomial Fitting
+
+For gamma correction formula, we take the logarithm:
+\f[
+ln(C_{sl})={\gamma}ln(C_s),\qquad C_s\ge0\
+\f]
+It can be seen that there is a linear relationship between \f$ln(C_s)\f$ and \f$ln(C_{sl})\f$. It can be considered that the formula is an approximation of a polynomial relationship, that is, there exists a polynomial $f$, which makes[2]:
+\f[
+ln(C_{sl})=f(ln(C_s)), \qquad C_s>0\\
+C_{sl}=0, \qquad C_s=0
+\f]
+
+Because \f$exp(ln(0))\to\infin\f$, the channel whose component is 0 is directly mapped to 0 in the formula above.
```

Comment:
Documentation is broken in this place (blames on undefined control sequence `\infin`)

Reply:
Thanks, I fixed this issue using \infty.